bin/src/upload.dart

```diff
@@ -27,7 +27,7 @@ class UploadPart extends CommandLinePart {
       });
     }
 
-    var pRoot = new Directory(res["package-root"]);
+    var pRoot = new File(res["packages"]);
```

Comment:
@Adracus @eseidel my focus was on getting `dart_coveralls` to run at all, as it currently only works with `coverage` versions prior to v0.7.4 (only `.packages` file allowed as input), which in turn will only work with SDK versions prior to v1.16.0.

Accordingly, backwards compatibility would have to introduce checks on SDK version and `coverage` package version, and combinations thereof, additionally depending on whether a file or dir was given as package source. Looks like quite some effort to get this right, especially since the `.packages` file is the way to go since v1.12.0. 

TL;DR: support for `package-root` dir requires an old version of `coverage` that doesn't work with Dart SDK >= 1.16.0. I don't think there's demand for this.